### ISSUE TITLE
Report error when binary/octal/hex number has no digits

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#43]
In debug builds, quick-lint-js crashes on the following JavaScript snippets:

```javascript
let x = 0x;
```
```javascript
let y = 0b;
```

This is because quick-lint-js assumes that a digit follows `0x` or `0b`, which is not necessarily true.

Teach quick-lint-js to report a user-friendly error on incomplete number literals. For example:
```
hello.js:1:9: binary number literal has no digits
```

Note: support for `0o`-prefixed octal number literals will be added in #7.

**Comment by vegerot** [comment]
See these tests from https://github.com/quick-lint/quick-lint-js/pull/40/commits/9982daef66bf41d4c4363697df270b8d560a8999#diff-163c0ffc2e0a69c625fe3df223fe7f20R272

```c++
TEST(test_lex, DISABLED_fail_lex_hex_numbers) {
  check_tokens(u8"0x42.3", {token_type::number, token_type::number});
  check_tokens(u8"0x.3", {token_type::number, token_type::number});
  check_tokens(u8"0x%", {token_type::number, token_type::number});
}
```

**Comment by strager** [comment]
@catac1 fixed this in commit d2d645a8d67055db88e5a6a027c3b0e4af76f30e.
